src/AzureExtension/DataModel/DataObjects/Organization.cs

```diff
@@ -1,134 +1,170 @@
-        }
-
-        return organization;
-    }
-
-    public static Organization GetOrCreate(DataStore dataStore, Uri connection)
-    {
-        var newOrganization = Create(connection);
-        return AddOrUpdateOrganization(dataStore, newOrganization);
-    }
-}
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Dapper;
+using Dapper.Contrib.Extensions;
+using DevHomeAzureExtension.Client;
+using DevHomeAzureExtension.Helpers;
+using Serilog;
+
+namespace DevHomeAzureExtension.DataModel;
+
+[Table("Organization")]
+public class Organization
+{
+    private static readonly Lazy<ILogger> _logger = new(() => Log.ForContext("SourceContext", $"DataModel/{nameof(Organization)}"));
+
+    private static readonly ILogger _log = _logger.Value;
+
+    // This is the time between seeing a potential updated Organization record and updating it.
+    // This value / 2 is the average time between Organization updating their Organization data and
+    // having it reflected in the datastore.
+    private static readonly long _updateThreshold = TimeSpan.FromHours(4).Ticks;
+
+    [Key]
+    public long Id { get; set; } = DataStore.NoForeignKey;
+
+    public string Name { get; set; } = string.Empty;
+
+    // Connection in string form. An organization may have multiple valid connection strings.
+    // Example: https://dev.azure.com/{organization} and https://{organization}.visualstudio.com
+    // We will treat the connection string as the unique identifier of the organization.
+    // This leaves open potential duplicate data, but not likely much.
+    // For this extension to function currently it does not matter.
+    public string Connection { get; set; } = string.Empty;
+
+    // When this record was updated.
+    public long TimeUpdated { get; set; } = DataStore.NoForeignKey;
+
+    // When all records related to this one (i.e. projects) were completely updated.
+    public long TimeLastSync { get; set; } = DataStore.NoForeignKey;
+
+    [Write(false)]
+    [System.Diagnostics.CodeAnalysis.SuppressMessage("CodeQuality", "IDE0052:Remove unread private members", Justification = "Will be used in near-future changes.")]
+    private DataStore? DataStore { get; set; }
+
+    [Write(false)]
+    [Computed]
+    public DateTime UpdatedAt => TimeUpdated.ToDateTime();
+
+    [Write(false)]
+    [Computed]
+    public DateTime LastSyncAt => TimeLastSync.ToDateTime();
+
+    [Write(false)]
+    [Computed]
+    public Uri ConnectionUri => new(Connection);
+
+    public override string ToString() => Name;
+
+    public void SetSynced()
+    {
+        TimeLastSync = DateTime.Now.ToDataStoreInteger();
+        if (DataStore?.Connection is not null)
+        {
+            DataStore.Connection.Update(this);
+        }
+    }
+
+    // Sets updated and sync time for all organization rows to 0, causing them to qualify for updating.
+    public static void ClearAllSynced(DataStore dataStore)

```

Comment:
Is the purpose of this method to clear all synched data, or to qualify the data for updating?

Reply:
It qualifies it for updating. It clears the updated and sync timestamps for the Organization objects, it does not delete the data (I removed the unintentional comment that mentioned delete). The next time an update is attempted all organizations will attempt to be updated.

Reply:
If the purpose of the method is to get all organizations in a state so every organization will attempt an update, why is it called `ClearAllSynched`?
What I'm getting at is this.  Should the method be named for what it actually does, or the purpose?

If we get a new hire 7 months down the road and they need to qualify all organizations to update, how will they know to call this?

Reply:
I renamed to "ClearAllSyncData", and I believe it to be sufficiently descriptive as to what it does.